src/Entities/Template/Forms/FormBuilder.jsx

```diff
@@ -0,0 +1,310 @@
+            style={{ order }}
+          >
+            <label className="fw-bold w-25">{label}</label>
+
+            <ValueView className={[viewClassName, "w-75"].join(" ")}>
+              {format !== "markdown" ? (
+                <span>
+                  {(fieldType === "array" && format === "comma-separated"
+                    ? fieldValue.filter((string) => string.length > 0).join(", ")
+                    : fieldValue
+                  )?.toString?.() || "none"}
+                </span>
+              ) : (fieldValue?.length ?? 0) > 0 ? (
+                <Widget
+                  src={"devhub.near/widget/devhub.components.molecule.MarkdownViewer"}
+                  props={{
+                    text: fieldValue,
+                  }}
+                />
+              ) : (
+                <span>none</span>
+              )}
+            </ValueView>
+          </div>
+
+          {renderInput(key, form, fieldSchema, fieldType, fieldValue, isEditable)}
+        </>
+      );
+    })}
+  </>
+);
+
+const Configurator = ({
+  actionsAdditional,
+  cancelLabel,
+  classNames,
+  externalState,
+  fieldsRender: customFieldsRender,
+  formatter: toFormatted,
+  isValid,
+  isActive,
+  onCancel,
+  onChange,
+  onSubmit,
+  schema,
+  submitIcon,
+  submitLabel,
+  hideSubmitBtn,
+}) => {
+  const fieldsRender = customFieldsRender || defaultFieldsRender;
+
+  const initialValues = Struct.typeMatch(schema) ? Struct.pick(externalState ?? {}, Object.keys(schema)) : {};
+
+  const form = useForm({ initialValues, onUpdate: onChange, stateKey: "form" });
+
+  const formFormattedValues = toFormatted ? toFormatted(form.values) : form.values;
+
+  const internalValidation = () =>
+    Object.keys(schema).every((key) => {
+      const fieldDefinition = schema[key];
+      const value = form.values[key];
+      if (!value || value.length === 0) {
+        return !fieldDefinition.inputProps.required;
+      } else if (fieldDefinition.inputProps.min && fieldDefinition.inputProps.min > value?.length) {
+        return false;
+      } else if (fieldDefinition.inputProps.max && fieldDefinition.inputProps.max < value?.length) {
+        return false;
+      } else if (fieldDefinition.inputProps.allowCommaAndSpace === false && /^[^,\s]*$/.test(value) === false) {
+        return false;
+      } else if (
+        fieldDefinition.inputProps.validUrl === true &&
+        /^(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)$/.test(value) === false
+      ) {
+        return false;
+      }
+      return true;
+    });
+
+  const isFormValid = () => {
+    return internalValidation() && (!isValid || isValid(formFormattedValues));
+  };
+
+  const onCancelClick = () => {
+    form.reset();
+    if (onCancel) onCancel();
+  };
+
+  const onSubmitClick = () => {
+    if (onSubmit && isFormValid()) {
+      onSubmit(formFormattedValues);
+    }
+  };
+
+  return (
+    <div className="flex-grow-1 d-flex flex-column gap-4">
```

Comment:
Just a heads up @gabehamilton, we're going to be moving away from Bootstrap with the new React on Chain engine. So let's try to stay away from using global Bootstrap classes like `d-flex`, etc.

Reply:
Cool. This component comes to us by way of DevHub. I was thinking it needs a rewrite anyway so I'll put in an Issue to move it off of bootstrap.